appmap/_implementation/recording.py

```diff
@@ -188,35 +189,46 @@ def is_member_func(m):
 
 
 class Recorder:
-    """ Singleton Recorder class """
-    _instance = None
+    
```

Comment:
Note
- in `appmap-ruby` Tracing contains many Tracers, one per thread.
- in `appmap-python` Recording contains one Recorder. The Recorder saves for many threads.

The two designs don't match.  I'm not suggesting changing the `appmap-python` design.

Is it ok for the two designs to not match?

Reply:
I don't think the designs need to match. Both seem clear to me, and the approach I took required the fewer changes to the source.

Reply:
I don't think the designs need to match. Both seem clear to me, and the approach I took required the fewer changes to the source.